pkgs/tools/security/rng-tools/default.nix

```diff
@@ -8,32 +8,43 @@
   # Not sure if jitterentropy is safe to use for cryptography
   # and thus a default entropy source
 , jitterentropy ? null, withJitterEntropy ? false
```

Comment:
We could add asserts for the dependencies of the optional features, so that if the feature is enabled the dependencies cannot be null, i.e. `withJitterEntropy -> jitterentropy != null`.
That's why I added the dependencies for the features in the same line as the option.

Reply:
I have seen that pattern in quite a few places but must admit I'm not really convinced about it. If you try building with a feature enabled but the relevant library is missing, the build will simply error out, so except for large packages where it could blow up 20 minutes into the build and we therefore want it to stop immediately, I don't really think we gain anything. That's just my opinion of course.

Reply:
Why would you not have the error upfront with nix instead of during the build?
So are you of the opinion we should only do this for large packages? Why not for small ones? What's the advantage?
Sorry, I don't quite understand what you're trying to say.